src/components/leaf.js

```diff
@@ -88,23 +88,16 @@ class Leaf extends React.Component {
     const text = this.renderText(props)
     if (el.textContent != text) return true
 
-    // If the selection was previously focused, and now it isn't, re-render so
-    // that the selection will be properly removed.
-    if (this.props.state.isFocused && props.state.isBlurred) {
-      const { index, node, ranges, state } = this.props
-      const { start, end } = OffsetKey.findBounds(index, ranges)
-      if (state.selection.hasEdgeBetween(node, start, end)) return true
-    }
-
-    // If the selection will be focused, only re-render if this leaf contains
-    // one or both of the selection's edges.
-    if (props.state.isFocused) {
+    // If the selection will be focused and native selection is not already
+    // in syncro with it, update native selection.
+    if (props.state.isFocused && props.selectionNeedsRedraw) {
       const { index, node, ranges, state } = props
       const { start, end } = OffsetKey.findBounds(index, ranges)
-      if (state.selection.hasEdgeBetween(node, start, end)) return true
+      if (state.selection.hasEdgeBetween(node, start, end)) {
+        this.updateSelection(props)
```

Comment:
Unfortunately we can't use `data.isNative`. :disappointed: 
First of all, `updateSelection` is triggered by selection changes that are generated programmatically and not by native selection changes. Indeed when the changes are consequences of native selection transformations the two selections are already in sync and there is no need to call 'updateSelection'.
Moreover setting `data.isNative` to `true` stops the updating of the whole editor content, but when selection changes, some components need re-rendering: for example `Void` `Inline` nodes which want to show an highlight border if selected.
The solution I adopted is to split in two the `Leaf` component.
The *higher* component, still named `Leaf`, is responsible for updating the native selection, the *lower*, named `LeafContent`, instead is used to render the leaf text content.
In this way we can update the `Leaf`, and therefore the native selection, without touching its content.